src/output/render-manager.cpp

```diff
@@ -377,6 +377,8 @@ class wf::render_manager::impl
             {
                 output->handle->frame_pending = true;
                 repaint_timer.set_timeout(total, [=] () {
+                    if (output->handle == nullptr)
```

Comment:
Is this necessary (i.e have you experienced a crash because of this)?

When the output is destroyed, its render_manager is also destroyed, leading to automatic disconnect of this timer IIRC.

Reply:
Haven't experienced a crash, just saw Sway doing this. And they do `wl_event_source_remove(output->repaint_timer)` in `output_destroy`..

Reply:
I think you can drop this check, output->handle is never supposed to get NULL. If that happens, then a much larger bug is there and Wayfire will crash in other places.